app/src/main/java/org/xbmc/kore/ui/sections/localfile/LocalPictureListFragment.java

```diff
@@ -0,0 +1,766 @@
+    public static final String ROOT_PATH_LOCATION = "rootPath";
+    public static final String DELAY_LOAD = "delayLoad";
+    private static final String ADDON_SOURCE = "addons:";
+
+    private String rootPathLocation = null;
+
+    private HostManager hostManager;
+    /**
+     * Handler on which to post RPC callbacks
+     */
+    private Handler callbackHandler = new Handler();
+
+    String mediaType = Files.Media.FILES;
+    ListType.Sort sortMethod = null;
+    String parentDirectory = null;
+    int playlistId = PlaylistType.MUSIC_PLAYLISTID;             // this is the ID of the music player
+//    private MediaPictureListAdapter adapter = null;
+    boolean browseRootAlready = false;
+    LocalPictureListFragment.FileLocation loadOnVisible = null;
+
+    ArrayList<FileLocation> rootFileLocation = new ArrayList<>();
+    Queue<FileLocation> mediaQueueFileLocation = new LinkedList<>();
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putString(MEDIA_TYPE, mediaType);
+        outState.putParcelable(SORT_METHOD, sortMethod);
+        try {
+            outState.putParcelableArrayList(PATH_CONTENTS, (ArrayList<FileLocation>) ((MediaPictureListAdapter) getAdapter()).getFileItemList());
+        } catch (NullPointerException npe) {
+            // adapter is null probably nothing was save in bundle because the directory is empty
+            // ignore this so that the empty message would display later on
+        }
+        outState.putParcelableArrayList(ROOT_PATH_CONTENTS, rootFileLocation);
+        outState.putBoolean(ROOT_VISITED, browseRootAlready);
+    }
+
+    @Override
+    protected RecyclerViewEmptyViewSupport.OnItemClickListener createOnItemClickListener() {
+        return new RecyclerViewEmptyViewSupport.OnItemClickListener() {
+            @Override
+            public void onItemClick(View view, int position) {
+                handleFileSelect(((MediaPictureListAdapter) getAdapter()).getItem(position));
+            }
+        };
+    }
+
+    @Override
+    protected RecyclerView.Adapter createAdapter() {
+        return new MediaPictureListAdapter(getActivity(), R.layout.grid_item_picture);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View root = super.onCreateView(inflater, container, savedInstanceState);
+        Bundle args = getArguments();
+        FileLocation rootPath = null;
+
+
+        if (ContextCompat.checkSelfPermission(this.getActivity(), Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                != PackageManager.PERMISSION_GRANTED) {
+            // Permission to read files was not granted,
+            // prompt the user for storage access permissions:
+            ActivityCompat.requestPermissions(getActivity(),
+                    new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},
```

Comment:
`WRITE_EXTERNAL_STORAGE` ? Don't you mean `READ_EXTERNAL_STORAGE`. That's what's on the manifest.